tatau_core/node/producer/producer_node.py

```diff
@@ -716,18 +716,20 @@ def _process_epoch_in_progress(self, task_declaration: TaskDeclaration):
                         if ta.train_data.is_master:
                             # train_result.current_iteration == finished iteration
                             # task_declaration.current_iteration == iteration in progress
-                            if (ta.train_result.current_iteration >= task_declaration.current_iteration and
-                                    ta.train_result.eval_results):
+                            if ta.train_result.current_iteration >= task_declaration.current_iteration:
                                 tr_history = ta.train_result.__class__.get_history(ta.train_result.asset_id, db=self.db,
```

Comment:
get_history is class method but you can use 
```suggestion
                                tr_history = ta.train_result.get_history(ta.train_result.asset_id, db=self.db,
```